src/store/safeAppsSlice.ts

```diff
@@ -22,15 +22,23 @@ export const safeAppsSlice = createSlice({
       state[chainId] ??= { pinned: [] }
       state[chainId].pinned = pinned
     },
+    setSafeApps: (state, { payload }: PayloadAction<SafeAppsState>) => {
+      // We much return as we are overwriting the entire state
+      return payload
```

Comment:
I would think `state = payload` would also overwrite the entire state. What is the difference to returning?

Reply:
Assigning uses a draft state via [Immer](https://immerjs.github.io/immer/) wheras returning uses the `payload` as is. Assigning _should_ work in theory but my debugging says otherwise 🤷‍♂️